apis/v1alpha1/types.go

```diff
@@ -107,10 +135,49 @@ type GeoLocationDetails struct {
 	SubdivisionCode *string `json:"subdivisionCode,omitempty"`
 }
 
+// A complex type that contains information about the health check.
+type HealthCheckConfig struct {
+	// A complex type that identifies the CloudWatch alarm that you want Amazon
+	// Route 53 health checkers to use to determine whether the specified health
+	// check is healthy.
+	AlarmIdentifier              *AlarmIdentifier `json:"alarmIdentifier,omitempty"`
+	ChildHealthChecks            []*string        `json:"childHealthChecks,omitempty"`
+	Disabled                     *bool            `json:"disabled,omitempty"`
+	EnableSNI                    *bool            `json:"enableSNI,omitempty"`
+	FailureThreshold             *int64           `json:"failureThreshold,omitempty"`
+	FullyQualifiedDomainName     *string          `json:"fullyQualifiedDomainName,omitempty"`
+	HealthThreshold              *int64           `json:"healthThreshold,omitempty"`
+	IPAddress                    *string          `json:"iPAddress,omitempty"`
+	InsufficientDataHealthStatus *string          `json:"insufficientDataHealthStatus,omitempty"`
+	Inverted                     *bool            `json:"inverted,omitempty"`
+	MeasureLatency               *bool            `json:"measureLatency,omitempty"`
+	Port                         *int64           `json:"port,omitempty"`
+	Regions                      []*string        `json:"regions,omitempty"`
+	RequestInterval              *int64           `json:"requestInterval,omitempty"`
+	ResourcePath                 *string          `json:"resourcePath,omitempty"`
+	RoutingControlARN            *string          `json:"routingControlARN,omitempty"`
+	SearchString                 *string          `json:"searchString,omitempty"`
+	Type                         *string          `json:"type_,omitempty"`
```

Comment:
We can override the Go json tag here, to allow users to pass `type` instead of `type_`. Checkout this example https://github.com/aws-controllers-k8s/eks-controller/blob/main/generator.yaml#L400-L403

Reply:
Done